http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,260 @@
+                .compose(req -> req.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(HttpVersion.HTTP_1_1, httpClientResponse.version());
+                            requestCheckpoint.flag();
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    requestCheckpoint.flag();
+                }).onFailure(Throwable::printStackTrace));
+
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("HttpClient Vertx HTTP/2 protocol test with ssl, alpn, trustAll true and verifyHost false")
+    public void httpClientVertxHttp2(Vertx vertx, VertxTestContext vertxTestContext) {
+        Checkpoint requestCheckpoint = vertxTestContext.checkpoint(2);
+        HttpClientOptions httpClientOptions = new HttpClientOptions();
+        httpClientOptions.setSsl(true)
+                .setUseAlpn(true)
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(httpClientOptions).build();
+
+        assertNotNull(vertx);
+        assertNotNull(httpClient);
+        vertxTestContext
+                .verify(() -> {
+                    httpClient.request(HttpMethod.GET, app.getPort(Protocol.HTTPS), baseUri.getHost(), BASE_ENDPOINT)
+                            .compose(req -> req.send()
+                                    .compose(httpClientResponse -> {
+                                        requestCheckpoint.flag();
+                                        assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                                        assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                                        return httpClientResponse.body();
+                                    }))
+                            .onSuccess(body -> {
+                                assertThat("Body response", body.toString().contains(GREETING));
+                                requestCheckpoint.flag();
+                            }).onFailure(throwable -> vertxTestContext.failNow(throwable.getMessage()));
+                });
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("HttpClient Vertx just with HTTP/2 protocol option set and default port")
+    void http2ProtocolTest(Vertx vertx, VertxTestContext vertxTestContext) {
+        HttpClientOptions httpClientOptions = new HttpClientOptions()
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(httpClientOptions).build();
```

Comment:
you create `httpClient` inside `BeforeEach` then here and then close it once in `AfterEach`, I suggest it must be one to one relationship (not create it in before each).